src/fields.rs

```diff
@@ -113,516 +138,168 @@ impl FromStr for FieldConversion {
 /// | "era"        |   `ToPrimitiveAndRequireString`   | undefined  |
 /// | "eraYear"    |     `ToIntegerWithTruncation`     | undefined  |
 /// | "timeZone"   |              `None`               | undefined  |
-#[derive(Debug)]
+#[derive(Debug, Default, Clone)]
 pub struct TemporalFields {
-    bit_map: FieldMap,
-    year: Option<i32>,
-    month: Option<i32>,
-    month_code: Option<TinyStr4>, // TODO: Switch to icu compatible value.
-    day: Option<i32>,
-    hour: i32,
-    minute: i32,
-    second: i32,
-    millisecond: i32,
-    microsecond: i32,
-    nanosecond: i32,
-    offset: Option<String>,    // TODO: Switch to tinystr?
-    era: Option<TinyStr16>,    // TODO: switch to icu compatible value.
-    era_year: Option<i32>,     // TODO: switch to icu compatible value.
-    time_zone: Option<String>, // TODO: figure out the identifier for TimeZone.
-}
-
-impl Default for TemporalFields {
-    fn default() -> Self {
-        Self {
-            bit_map: FieldMap::empty(),
-            year: None,
-            month: None,
-            month_code: None,
-            day: None,
-            hour: 0,
-            minute: 0,
-            second: 0,
-            millisecond: 0,
-            microsecond: 0,
-            nanosecond: 0,
-            offset: None,
-            era: None,
-            era_year: None,
-            time_zone: None,
-        }
-    }
+    properties: FxHashMap<FieldKey, FieldValue>,
```

Comment:
Overall question. Why do we need to use a hashmap? I would say having a map-like interface for the current struct approach would also work well, with the advantage of being all in the stack.

Reply:
Hmmmm, changing the interface might work as well.

I'm not entirely sure how much of a fan I am of the large struct that this switches away from, but it could also primarily be the API for the struct that I'm annoyed by.

I'll play around with it a bit more :smile: